### PR TITLE
containerd: migrate to using buildGoPackage

### PR DESCRIPTION
**Pull request by vdemeester**
###### Motivation for this change

Part of #52469, make the `runc` derivation using `buildGoPackage` :angel: 

… and add man pages, which means `containerd` becomes a multi-output
derivation : `containerd.bin` and `containerd.man`.

cc @Mic92 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/virtualization/containerd/default.nix [hunk 1]
```diff
@@ -14,34 +13,37 @@ stdenv.mkDerivation rec {
...
+    done
+
+    make man
+    manRoot="$man/share/man"
+    mkdir -p "$manRoot"
+    for manFile in man/*; do
+      manName="$(basename "$manFile")" # "docker-build.1"
+      number="$(echo $manName | rev | cut -d'.' -f1 | rev)"
+      mkdir -p "$manRoot/man$number"
+      gzip -c "$manFile" > "$manRoot/man$number/$manName.gz"
```
Comment (Mic92) [inline pkgs/applications/virtualization/containerd/default.nix]: I think we automatically compress manpages.

Comment (vdemeester) [reply pkgs/applications/virtualization/containerd/default.nix]: oh :thinking: then there is some other packages that should change too

**Comment by vdemeester** [comment]
oh this broke the docker derivation :sweat_smile: I'll fix :angel:

**Comment by Mic92** [comment]
@GrahamcOfBorg test docker

**Comment by Mic92** [comment]
@GrahamcOfBorg build nixosTests.docker

**Pull request merged**